pkg/apis/application/v1alpha1/app_project_types.go

```diff
@@ -406,18 +406,33 @@ func (proj AppProject) IsGroupKindPermitted(gk schema.GroupKind, namespaced bool
 		namespaceBlacklist := proj.Spec.NamespaceResourceBlacklist
 
 		isWhiteListed = namespaceWhitelist == nil || len(namespaceWhitelist) != 0 && isResourceInList(res, namespaceWhitelist)
-		isBlackListed = len(namespaceBlacklist) != 0 && isResourceInList(res, namespaceBlacklist)
+		isBlackListed = len(namespaceBlacklist) != 0 && isResourceInBlacklist(res, namespaceBlacklist)
 		return isWhiteListed && !isBlackListed
 	}
 
 	clusterWhitelist := proj.Spec.ClusterResourceWhitelist
 	clusterBlacklist := proj.Spec.ClusterResourceBlacklist
 
 	isWhiteListed = len(clusterWhitelist) != 0 && isResourceInList(res, clusterWhitelist)
-	isBlackListed = len(clusterBlacklist) != 0 && isResourceInList(res, clusterBlacklist)
+	isBlackListed = len(clusterBlacklist) != 0 && isResourceInBlacklist(res, clusterBlacklist)
 	return isWhiteListed && !isBlackListed
 }
 
+// IsGroupKindVisible validates if the given resource group/kind is visible in the project
+func (proj AppProject) IsGroupKindVisible(gk schema.GroupKind, namespaced bool) bool {
+	permitted := proj.IsGroupKindPermitted(gk, namespaced)
+	res := metav1.GroupKind{Group: gk.Group, Kind: gk.Kind}
```

Comment:
this assignment can be moved after line `427` because it's useless if `permitted` is `true`